src/dmd/func.d

```diff
@@ -3150,6 +3197,60 @@ private bool checkEscapingSiblings(FuncDeclaration f, FuncDeclaration outerFunc,
     return bAnyClosures;
 }
 
+/***
+ * Returns true if any of the symbols `p` resides in the enclosing instantiation scope of `s`.
+ */
+bool followInstantiationContext(D...)(Dsymbol s, D p)
+{
+    static bool has2This(Dsymbol s)
+    {
+        if (auto f = s.isFuncDeclaration())
+            return f.vthis2 !is null;
+        if (auto ad = s.isAggregateDeclaration())
+            return ad.vthis2 !is null;
+        return false;
+    }
+
+    assert(s);
+    if (has2This(s))
+    {
+        assert(p.length);
+        auto parent = s.toParent();
+        while (parent)
+        {
+            auto ti = parent.isTemplateInstance();
+            if (!ti) break;
+            foreach (oarg; *ti.tiargs)
+            {
+                auto sa = getDsymbol(oarg);
+                if (!sa)
+                    continue;
+                sa = sa.toAlias().toParent2();
+                if (!sa)
+                    continue;
+                foreach (ps; p)
+                {
+                    if (sa == ps)
+                        return true;
+                }
+            }
+            parent = ti.tempdecl.toParent();
+        }
+        return false;
+    }
+    return false;
+}
+
+/*
+ * Returns the declaration scope scope of `s`
+ * unless any of the symbols `p` resides in its enclosing instantiation scope
+ * then the latter is returned.
+ */
+Dsymbol toParentP(D...)(Dsymbol s, D p)
```

Comment:
is this other variadic template really needed?

Reply:
no but see my reply the other similar case.